src/components/popover-container/popover-container.pw.tsx

```diff
@@ -0,0 +1,683 @@
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for Title example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Title />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for Position example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Position />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for CoverButton example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<CoverButton />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for RenderProps example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<RenderProps />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for Controlled example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Controlled />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for Complex example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Complex />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for Filter example", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Filter />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+
+    await checkAccessibility(page);
+  });
+
+  test("should check accessibility for Filter example with filter button clicked", async ({
+    mount,
+    page,
+  }) => {
+    await mount(<Filter />);
+
+    const popoverSettingsIconElement = popoverSettingsIcon(page);
+    await popoverSettingsIconElement.click();
+    const checkboxElementOne = page.locator(`[role="checkbox"]`).nth(0);
+    const checkboxElementTwo = page.locator(`[role="checkbox"]`).nth(1);
+    const checkboxElementThree = page.locator(`[role="checkbox"]`).nth(2);
+
+    await checkboxElementOne.check();
+    await checkboxElementTwo.check();
+    await checkboxElementThree.check();
+    const mainTextElement = getDataElementByValue(page, "main-text").nth(1);
+    await mainTextElement.click();
+    const pillComponentOne = getComponent(page, "pill").nth(0);
```

Comment:
**suggestion**: we could run these pill checks in parallel rather than in sequence, to speed up the test (we don't do any actions in between so these should all be independent of each other). Something like:

```
const assertions = [0, 1 , 2].map(index => expect(getComponent(page, "pill").nth(index)).toBeVisible();
await Promise.all(assertions);
```

Reply:
Big brain thinking 🧠 
Good idea, I'll get it changed to that 👍🏻